Fix license information

anthonyfok opened the pull request:
There appear to be some conflicting information about ibus-array's licensing:

1. https://code.google.com/archive/p/ibus-array/ says it is GNU GPL v2, linking to https://www.gnu.org/licenses/old-licenses/gpl-2.0.html, corresponding to SPDX-License-Identifier: GPL-2.0-or-later
2. The COPYING file is a copy of GPL-3.0-or-later
3. The source files that have the GPL info in the header, all are referring to GPL-2.0-or-later, although they all have the old FSF address listed.
4. (Debian package) debian/copyright says most files are GPL-2.0-only, but none of the source files says so.

With the above information, I think we can infer that Yu-Chun Wang originally intended the project to be licensed under GPL-2.0-or-later, but goofed when he placed a GPL-3.0-or-later COPYING file by mistake, hence this PR to fix COPYING and to update FSF address.

lexical commented:
I just mail the original author, waiting for the reply.

mainlander commented:
I am the original author, Yu-Chun Wang. ibus-array was released under GPL v2. Please help correct the COPYING file and the headers of sources. Thanks very much.

lexical commented:
@mainlander, which did you prefer ibus-array's original code to be GPL-2.0-or-later or GPL-2.0-only?

mainlander commented:
I prefer GPL-2.0-or-later. Many thanks.

lexical commented:
@mainlander Thanks for the reply.
@anthonyfok I believe you that you also agree with the re-licensing, right? Since you sent the PR.

There are still a few contributors to contact:
Author: Danilo Chang <ray2501@gmail.com>
Author: OBATA Akio <obache@outlook.com>
Author: slyfox3 <innocentfox@gmail.com>

anthonyfok commented:
> @mainlander Thanks for the reply.
> @anthonyfok I believe you that you also agree with the re-licensing, right? Since you sent the PR.

Yes, since I submitted the PR in the first place, I agree with merging my PR in order to resolve the issue.  ;-)

I would not call it "re-licensing", per se, because the the original author @mainlander has intended to use GPL-2.0-or-later since the intial commit in 2009.  This is especially clearly stated in ibus-array.spec.in https://github.com/lexical/ibus-array/blob/5a2b59e2c702a3099bd8ff26ae866d10bf9104f7/ibus-array.spec.in where it says `License: GPLv2+`.

> There are still a few contributors to contact:
> Author: Danilo Chang [ray2501@gmail.com](mailto:ray2501@gmail.com)

* Commit 7eefe89:
** changed src/engine.c, GPL-2.0-or-later copyright header was already in the file.

> Author: OBATA Akio [obache@outlook.com](mailto:obache@outlook.com)

OBATA Akio contributed two commits:

Commit 8c062b2:
* changed setup/main.py, GPL-2.0-or-later copyright header was already in the file

Commit 8f00cfba:
* changed setup/ibus-setup-array.in, GPL-2.0-or-later copyright header was already in the file.

> Author: slyfox3 [innocentfox@gmail.com](mailto:innocentfox@gmail.com)

Commits d20ca12, d045460
* changed data/array30_simplecode.cin, which, though unstated, should fall under the "Array Input Method Public License" while being compatible with GPL, see https://fedoraproject.org/wiki/Licensing/Array

Commit: 65c926b
* changed INSTALL, which is licensed under the FSF Unlimited License, see https://spdx.org/licenses/FSFUL.html

So, I don't think any of these contributors have seen any sort of "licensing change" or "license confusion", thus I think there is really no need to contact them for their consent.
